explore_data/cluster_by_accessory_gene.py

```diff
@@ -0,0 +1,609 @@
+)
+fig3 += pn.geom_point(pn.aes(color="Strain type_pao1"), alpha=0.4)
+fig3 += pn.labs(
+    x="median expression of PAO1-only genes (TPM)",
+    y="median expression of PA14-only genes (TPM)",
+    title="TPM of accessory genes",
+)
+fig3 += pn.theme_bw()
+fig3 += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=10),
+    axis_title=pn.element_text(family="sans-serif", size=12),
+)
+fig3 += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+print(fig3)
+
+# +
+# Plot log-scaled
+fig4 = pn.ggplot(
+    pao1_pa14_acc_expression_label,
+    pn.aes(x="median acc expression_pao1", y="median acc expression_pa14"),
+)
+fig4 += pn.scales.scale_x_log10()
+fig4 += pn.scales.scale_y_log10()
+fig4 += pn.geom_point(pn.aes(color="Strain type_pao1"), alpha=0.4)
+fig4 += pn.labs(
+    x="median expression of PAO1-only genes (log10 TPM)",
+    y="median expression of PA14-only genes (log10 TPM)",
+    title="TPM of accessory genes",
+)
+fig4 += pn.theme_bw()
+fig4 += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=10),
+    axis_title=pn.element_text(family="sans-serif", size=12),
+)
+fig4 += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+
+print(fig4)
+# -
+
+# **Note:**
+# We can compare TPM between PAO1 and PA14 if they have similar total number of reads mapped. So perhaps we can compare PAO1 and PA14 compendia. See Georgia’s [google doc](https://docs.google.com/document/d/1OnvJNVkhK5ATnhHaeSmz8Ze_Iw9ixXhfSxAL_BSqbuM/edit)
+
+# **Takeaway:**
+# * This is a positive control that PAO1 annotated samples have higher median expression of PAO1-only genes compared to PA14-only genes. And similarly, PA14 annotated samples have higher median expression of PA14-only genes compared to PAO1-only genes. In other words, we expect that PA14-only genes will have either 0 or very low values in PAO1 samples and vice versa.
+#
+# * This result also shows that we can anticipate a very clear binning of our samples into PAO1 and PA14 if we use mapping rates.
+#
+# * Note: The NA strains are those where the strain information was not available in the metadata. By a quick manual spot check it looks like a bunch were clinical isolates (which is good since these NA seem to cluster with other clinical isolates).
```

Comment:
Why is the spread of expression values between PAO1 and PA14 different?

Reply:
Good question. PAO1 higher median accessory expression than PA14. My initial guess that that maybe the PAO1 reference had higher mapping rates however, it looks like they have similar distributions of total reads mapped and mapping rate. So I'll have to think about what else might be causing this difference in range.

Reply:
The good thing is that for the planned downstream analysis we're planning to separate between PAO1 and PA14 so this range shouldn't cause an issue in that sense